src/components/MasterDetails.js

```diff
@@ -155,32 +160,37 @@ export default class MasterDetails extends React.Component<MasterDetailsDefaultP
   constructor(props: MasterDetailsProps) {
     super(props);
     this.state = {
-      selectedRows: this.props.rows.length > 0 ? [this.props.rows[0].id] : [],
-      detailsRowId: this.props.rows.length > 0 ? this.props.rows[0].id : '',
+      selectedRows: props.rows.length > 0 && props.rows[0] ? [props.rows[0].id] : [],
+      detailsRowId: props.rows.length > 0 && props.rows[0] ? props.rows[0].id : '',
     };
     (this: any).selectionChanged = this.selectionChanged.bind(this);
   }
   state: MasterDetailsState;
 
   componentWillReceiveProps(newProps: MasterDetailsProps) {
-    if (newProps.rows.length === 0) {
-      this.setState({
-        selectedRows: [],
-        detailsRowId: '',
-      });
-    } else {
-      const ids = newProps.rows.map((row) => {
-        return row.id;
-      });
-      const newSelectedRows = this.state.selectedRows.filter((row) => { return ids.includes(row); });
-      if (this.props.noEmptySelection && newSelectedRows.length === 0) {
-        newSelectedRows.push(newProps.rows[0].id);
+    const rowsChanged = !ObjectUtils.arrayEquals(newProps.rows, this.props.rows);
+
+    if (rowsChanged) {
+      if (newProps.rows.length === 0) {
+        this.setState({
+          selectedRows: [],
+          detailsRowId: null,
```

Comment:
Does Flow not complain about setting detailsRowId to null when the type is specified as `string` in MasterDetailsState?

Reply:
I'm not getting inline flow feedback, but flow did yell at me with the command line check - fixing it.